src/mito2/src/worker.rs

```diff
@@ -579,7 +580,7 @@ type RequestBuffer = Vec<WorkerRequest>;
 #[derive(Default)]
 pub(crate) struct StalledRequests {
     /// Stalled requests.
-    pub(crate) requests: Vec<SenderWriteRequest>,
+    pub(crate) requests: HashMap<RegionId, Vec<SenderWriteRequest>>,
```

Comment:
`HashMap<RegionId, (estimated_size: usize, Vec<SenderWriteRequest>)>,`

We can avoid recalculating `estimated_size` when the remove operation occurs.

Reply:
Maybe we can store the estimated size for each region.